include/BingoCpp/implicit_regression.h

```diff
@@ -0,0 +1,46 @@
+#ifndef BINGOCPP_INCLUDE_BINGOCPP_IMPLICIT_REGRESSION_H_
+#define BINGOCPP_INCLUDE_BINGOCPP_IMPLICIT_REGRESSION_H_
+
+#include <Eigen/Dense>
+
+#include "BingoCpp/equation.h"
+#include "BingoCpp/fitness_function.h"
+#include "BingoCpp/training_data.h"
+
+namespace bingo {
+
+struct ImplicitTrainingData : TrainingData {
+ public:
+  Eigen::ArrayXXd x;
+  Eigen::ArrayXXd dx_dt;
+  ImplicitTrainingData(const Eigen::ArrayXXd &input);
+  ImplicitTrainingData(const Eigen::ArrayXXd &input,
+                       const Eigen::ArrayXXd &derivative);
+  ImplicitTrainingData* GetItem(int item);
+  ImplicitTrainingData* GetItem(const std::vector<int> &items);
+  int Size() { return x.rows(); }
+};
+
+class ImplicitRegression : public VectorBasedFunction {
```

Comment:
Could we expose the metric parameter to the VectorBasedFunction like we did in the Explicit regression?